src/manager/runner_manager.py

```diff
@@ -0,0 +1,340 @@
+        )
+
+        runner_names = cloud_infos_map.keys() & github_infos_map.keys()
+        cloud_only = cloud_infos_map.keys() - runner_names
+        github_only = github_infos_map.keys() - runner_names
+        if cloud_only:
+            logger.warning(
+                "Found runner instance on cloud but not registered on GitHub: %s", cloud_only
+            )
+        if github_only:
+            logger.warning(
+                "Found self-hosted runner on GitHub but no matching runner instance on cloud: %s",
+                github_only,
+            )
+
+        runner_instances: list[RunnerInstance] = [
+            RunnerInstance(
+                cloud_infos_map[name], github_infos_map[name] if name in github_infos_map else None
+            )
+            for name in cloud_infos_map.keys()
+        ]
+        if cloud_runner_state is not None:
+            runner_instances = [
+                runner for runner in runner_instances if runner.cloud_state in cloud_runner_state
+            ]
+        if github_runner_state is not None:
+            runner_instances = [
+                runner
+                for runner in runner_instances
+                if runner.github_state is not None and runner.github_state in github_runner_state
+            ]
+        return cast(tuple[RunnerInstance], tuple(runner_instances))
+
+    def delete_runners(self, num: int) -> IssuedMetricEventsStats:
+        """Delete runners.
+
+        Args:
+            num: The number of runner to delete.
+
+        Returns:
+            Stats on metrics events issued during the deletion of runners.
+        """
+        logger.info("Deleting %s number of runners", num)
+        runners_list = self.get_runners()[:num]
+        runner_names = [runner.name for runner in runners_list]
+        logger.info("Deleting runners: %s", runner_names)
+        remove_token = self._github.get_removal_token()
+        return self._delete_runners(runners=runners_list, remove_token=remove_token)
+
+    def flush_runners(
+        self, flush_mode: FlushMode = FlushMode.FLUSH_IDLE
+    ) -> IssuedMetricEventsStats:
+        """Delete runners according to state.
+
+        Args:
+            flush_mode: The type of runners affect by the deletion.
+
+        Returns:
+            Stats on metrics events issued during the deletion of runners.
+        """
+        match flush_mode:
+            case FlushMode.FLUSH_IDLE:
+                logger.info("Flushing idle runners...")
+            case FlushMode.FLUSH_BUSY:
+                logger.info("Flushing idle and busy runners...")
+            case _:
+                logger.critical(
+                    "Unknown flush mode %s encountered, contact developers", flush_mode
+                )
+
+        states = [GithubRunnerState.IDLE]
+        if flush_mode == FlushMode.FLUSH_BUSY:
+            states.append(GithubRunnerState.BUSY)
+
+        runners_list = self.get_runners(github_runner_state=states)
```

Comment:
Should both be singular or plural for state?

Reply:
And perhaps remove the runner namespace from the arg

Reply:
Fixed